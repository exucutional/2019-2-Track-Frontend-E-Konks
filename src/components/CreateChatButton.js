/* eslint-disable react/destructuring-assignment */
import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { keyframes } from '@emotion/core';

const Container = styled.div``;

const Pulse = keyframes`
	0% {
		transform: scale(1, 1);
	}
	50% {
		transform: scale(1.1, 1.1);
	}
	100% {
		transform: scale(1, 1);
	}
`;

const Button = styled.img`
	width: 10vh;
	position: fixed;
	right: 20px;
	bottom: 20px;
	&:hover {
		cursor: pointer;
		animation: ${Pulse} 2s infinite ease-in-out;
	};
	cursor: pointer;
`;

function CreateChatButton(props) {
	return (
		<Container>
			<Button
				src="https://image.flaticon.com/icons/svg/1159/1159633.svg"
				id="create-chat-button"
				onClick={props.onClick}
			/>
		</Container>
	);
}

CreateChatButton.propTypes = {
	onClick: PropTypes.func.isRequired,
};

export default CreateChatButton;
