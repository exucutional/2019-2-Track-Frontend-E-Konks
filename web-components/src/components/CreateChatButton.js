/* eslint-disable linebreak-style */
/* eslint-disable class-methods-use-this */
/* eslint-disable linebreak-style */
import './CreateChatForm';

const template = document.createElement('template');
template.innerHTML = `
    <style>
    .create-chat-button-obj {
        width: 2em;
        pointer-events: none;
    }

    .create-chat-button {
        width: 2em;
        position: fixed;
        right: 20px;
        bottom: 20px;
    }

    create-chat-form {
        display: none;
        margin: 1em;
        border: solid;
        width: fit-content;
    }
    </style>
    <form>
        <create-chat-form></create-chat-form>
        <div class='create-chat-button'>
            <object
                class="create-chat-button-obj"
                type="image/svg+xml"
                data="data/pen.svg">
                <img src="data/pen.svg">
            </object>
        </div>
    </form
`;

class CreateChatButton extends HTMLElement {
  constructor() {
    super();
    this._shadowRoot = this.attachShadow({ mode: 'open' });
    this._shadowRoot.appendChild(template.content.cloneNode(true));

    this.$form = this._shadowRoot.querySelector('form');
    this.$createChatForm = this._shadowRoot.querySelector('create-chat-form');
    this.$createChatButton = this._shadowRoot.querySelector('.create-chat-button');
    this.$createChatButton.addEventListener('click', this._onClickCreateChat.bind(this));
    this.$createChatForm.addEventListener('keypress', this._onKeyPress.bind(this));
  }

  _onClickCreateChat(event) {
    event.preventDefault();
    this.$createChatForm.style.display = 'flex';
    document.querySelector('html').scrollTop += document.querySelector('html').scrollHeight;
  }

  _onKeyPress(event) {
    if (event.keyCode === 13) {
      event.preventDefault();
      this.$createChatForm.style.display = 'none';
      this.dispatchEvent(new Event('submit'));
    }
  }

  get chatName() {
    return this.$createChatForm.value;
  }
}

customElements.define('create-chat-button', CreateChatButton);
