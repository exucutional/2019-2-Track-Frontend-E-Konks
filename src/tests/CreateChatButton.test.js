import React from 'react';
import renderer from 'react-test-renderer';
import CreateChatButton from '../components/CreateChatButton'

it('renders correctly', () => {
	const tree = renderer
		.create(<CreateChatButton onClick={ () => {} }/>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

