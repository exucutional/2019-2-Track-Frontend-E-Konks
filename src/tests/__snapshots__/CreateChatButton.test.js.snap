// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-0"
>
  <img
    className="css-1qz7riz"
    onClick={[Function]}
    src="https://image.flaticon.com/icons/svg/1159/1159633.svg"
  />
</div>
`;
